### DISCUSSION TITLE
How to use a '.wasm' module/file in moonbit main/lib?

### DISCUSSION THREAD

**Discussion by bikallem** [moonbitlang/moon#342] [General]
I have `lib.wasm` file compiled by zig. It exports two functions `add and mult`. I want to now use this in moonbit. How do I go about doing this in moonbit? The doc mentions using .wasm modules in via javascript. However, I want to be able to use it from moonbit. Is this possible?

**Comment by lijunchen** [comment]
No. Currently, we can only use inline wasm like:

```
extern "wasm" fn add(x : Int, y : Int) -> Int =
  #|(func (param $x i32) (param $y i32) (result i32)
  #|  (i32.add (local.get $x) (local.get $y))
  #|)

test {
  inspect!(add(1, 2), content="3")
}
```

**Reply by bikallem** [reply]
Thanks.
